log/loggers.go

```diff
@@ -5,122 +5,244 @@ import (
+	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil || !fields.debug {
+	if fields == nil {
 		return
 	}
-
-	displayError(fields.logger.newLogEvent(fmt.Sprintln(v...),
-		fields.logger.DebugHeader,
-		fields.name,
-		fields.output))
+	if fields.debug {
+		fields.output.StageLogEvent(func() string { return fmt.Sprintln(v...) },
+			fields.logger.DebugHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Debugf takes a pointer subLogger struct, string & interface formats and sends to Info()
+// Debugf takes a pointer subLogger struct, string & interface formats
 func Debugf(sl *SubLogger, data string, v ...interface{}) {
-	Debug(sl, fmt.Sprintf(data, v...))
+	mu.RLock()
+	defer mu.RUnlock()
+	fields := sl.getFields()
+	if fields == nil {
+		return
+	}
+	if fields.debug {
+		fields.output.StageLogEvent(func() string { return fmt.Sprintf(data, v...) },
+			fields.logger.DebugHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Warn takes a pointer subLogger struct & string  and sends to newLogEvent()
+// Warn takes a pointer subLogger struct & string and sends to StageLogEvent
 func Warn(sl *SubLogger, data string) {
+	mu.RLock()
+	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil || !fields.warn {
+	if fields == nil {
 		return
 	}
-	displayError(fields.logger.newLogEvent(data,
-		fields.logger.WarnHeader,
-		fields.name,
-		fields.output))
+	if fields.warn {
+		fields.output.StageLogEvent(func() string { return data },
+			fields.logger.WarnHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Warnln takes a pointer subLogger struct & interface formats and sends to newLogEvent()
+// Warnln takes a pointer subLogger struct & interface formats and sends to StageLogEvent
 func Warnln(sl *SubLogger, v ...interface{}) {
+	mu.RLock()
+	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil || !fields.warn {
+	if fields == nil {
 		return
 	}
-	displayError(fields.logger.newLogEvent(fmt.Sprintln(v...),
-		fields.logger.WarnHeader,
-		fields.name,
-		fields.output))
+	if fields.warn {
+		fields.output.StageLogEvent(func() string { return fmt.Sprintln(v...) },
+			fields.logger.WarnHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Warnf takes a pointer subLogger struct, string & interface formats and sends to Warn()
+// Warnf takes a pointer subLogger struct, string & interface formats
```

Comment:
inconsistent comment style
```suggestion
// Warnf takes a pointer subLogger struct, string and interface formats sends to StageLogEvent
```